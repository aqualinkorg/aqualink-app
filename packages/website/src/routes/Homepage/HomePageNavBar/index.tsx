import React from "react";
import { AppBar, Toolbar, Grid, Button } from "@material-ui/core";

const HomePageNavBar = () => {
  return (
    <AppBar position="static">
      <Toolbar>
        <Grid item xs={12}>
          <Grid container justify="flex-end" alignItems="center">
            <Grid item>
              <Button variant="contained" color="secondary">
                Log In
              </Button>
            </Grid>
          </Grid>
        </Grid>
      </Toolbar>
    </AppBar>
  );
};

export default HomePageNavBar;
