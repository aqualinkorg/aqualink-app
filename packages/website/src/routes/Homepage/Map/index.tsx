import React, { useState, useRef, useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { Map, TileLayer, Marker, Popup as LeafletPopup } from "react-leaflet";
import L from "leaflet";
import {
  createStyles,
  withStyles,
  WithStyles,
  CircularProgress,
} from "@material-ui/core";

import {
  reefsListSelector,
  reefsListLoadingSelector,
} from "../../../store/Reefs/reefsListSlice";
import {
  reefOnMapSelector,
  unsetReefOnMap,
} from "../../../store/Homepage/homepageSlice";
import { Reef } from "../../../store/Reefs/types";
import Popup from "./Popup";
import { coloredBuoy } from "./utils";
import {
  colorFinder,
  degreeHeatingWeeksCalculator,
} from "../../../helpers/degreeHeatingWeeks";

const HomepageMap = ({ classes }: HomepageMapProps) => {
  const mapRef = useRef<Map>(null);
  const markerRef = useRef<Marker>(null);
  const dispatch = useDispatch();
  const reefOnMap = useSelector(reefOnMapSelector);
  const reefsList = useSelector(reefsListSelector);
  const loading = useSelector(reefsListLoadingSelector);
  const [center, setCenter] = useState<[number, number]>([0, 0]);
  const [zoom, setZoom] = useState<number>(2);

  useEffect(() => {
    const { current } = mapRef;
    if (current && current.leafletElement) {
      const map = current.leafletElement;
      const southWest = L.latLng(-90, -240);
      const northEast = L.latLng(90, 240);
      const bounds = L.latLngBounds(southWest, northEast);
      map.setMaxBounds(bounds);
      map.flyTo(center, zoom, { duration: 1 });
    }
  }, [center, zoom, mapRef]);

  // Add Sofar Tiles
  useEffect(() => {
    const { current } = mapRef;
    if (current && current.leafletElement) {
      const map = current.leafletElement;

      const layerControl = L.control.layers(undefined).addTo(map);
      const sofarLayers = [
        {
          name: "Sea Surface Temperature",
          model: "HYCOM",
          variableID: "seaSurfaceTemperature",
          cmap: "turbo",
        },
        {
          name: "NOAA Degree Heating Week",
          model: "NOAACoralReefWatch",
          variableID: "degreeHeatingWeek",
          cmap: "noaacoral",
        },
      ];

      const sofarUrl = "https://api.sofarocean.com/marine-weather/v1/models/";
      const { REACT_APP_SOFAR_API_TOKEN: token } = process.env;
      sofarLayers.forEach((layer) => {
        layerControl.addOverlay(
          L.tileLayer(
            `${sofarUrl}${layer.model}/tile/{z}/{x}/{y}.png?colormap=${layer.cmap}&token=${token}&variableID=${layer.variableID}`,
            { opacity: 0.5 }
          ),
          layer.name
        );
      });
    }
  }, [loading]);

  useEffect(() => {
    const markerCurrent = markerRef.current;
    const mapCurrent = mapRef.current;
    if (
      markerCurrent &&
      markerCurrent.leafletElement &&
      mapCurrent &&
      mapCurrent.leafletElement &&
      reefOnMap?.polygon?.type === "Point"
    ) {
      const map = mapCurrent.leafletElement;
      const marker = markerCurrent.leafletElement;
      map.flyTo(
        [reefOnMap.polygon.coordinates[1], reefOnMap.polygon.coordinates[0]],
        6,
        { duration: 1 }
      );
      marker.openPopup();
    }
  }, [reefOnMap, markerRef, mapRef]);

  return (
    <>
      {loading ? (
        <div className={classes.loading}>
          <CircularProgress size="10rem" thickness={1} />
        </div>
      ) : (
        <Map
          maxBoundsViscosity={1.0}
          className={classes.map}
          ref={mapRef}
          center={center}
          zoom={zoom}
          minZoom={2}
        >
          <TileLayer url="https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}" />
          {reefOnMap && reefOnMap.polygon.type === "Point" && (
            <Marker
              ref={markerRef}
              icon={L.divIcon({
                iconSize: [24, 24],
                iconAnchor: [8, 48],
                popupAnchor: [3, -48],
                html: `${coloredBuoy(
                  colorFinder(
                    degreeHeatingWeeksCalculator(
                      reefOnMap.latestDailyData.degreeHeatingDays
                    )
                  )
                )}`,
                className: "marker-icon",
              })}
              position={[
                reefOnMap.polygon.coordinates[1],
                reefOnMap.polygon.coordinates[0],
              ]}
            >
              <LeafletPopup closeButton={false} className={classes.popup}>
                <Popup reef={reefOnMap} />
              </LeafletPopup>
            </Marker>
          )}
          {reefsList.length > 0 &&
            reefsList.map((reef: Reef) => {
              if (reef.polygon.type === "Point") {
                const [lng, lat] = reef.polygon.coordinates;
                const { degreeHeatingDays } = reef.latestDailyData || {};
                const color = colorFinder(
                  degreeHeatingWeeksCalculator(degreeHeatingDays)
                );
                return (
                  <Marker
                    onClick={() => {
                      setZoom(6);
                      setCenter([lat, lng]);
                      dispatch(unsetReefOnMap());
                    }}
                    key={reef.id}
                    icon={L.divIcon({
                      iconSize: [24, 24],
                      iconAnchor: [8, 48],
                      popupAnchor: [3, -48],
                      html: `${coloredBuoy(color)}`,
                      className: "marker-icon",
                    })}
                    position={[
                      reef.polygon.coordinates[1],
                      reef.polygon.coordinates[0],
                    ]}
                  >
                    <LeafletPopup closeButton={false} className={classes.popup}>
                      <Popup reef={reef} />
                    </LeafletPopup>
                  </Marker>
                );
              }
              return null;
            })}
        </Map>
      )}
    </>
  );
};

const styles = () =>
  createStyles({
    map: {
      height: "100%",
      width: "100%",
    },
    popup: {
      width: "12vw",
    },
    loading: {
      height: "100%",
      width: "100%",
      display: "flex",
      justifyContent: "center",
      alignItems: "center",
    },
  });

type HomepageMapProps = WithStyles<typeof styles>;

export default withStyles(styles)(HomepageMap);
