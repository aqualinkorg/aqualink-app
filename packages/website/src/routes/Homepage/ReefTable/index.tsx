import React, { useState } from "react";
import {
  withStyles,
  WithStyles,
  createStyles,
  Grid,
  Typography,
  IconButton,
  Hidden,
  TableContainer,
  Table,
  CircularProgress,
  Box,
  Theme,
} from "@material-ui/core";
import ArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import ArrowDownwardIcon from "@material-ui/icons/ArrowDownward";

import { useSelector } from "react-redux";
import SelectedReefCard from "./SelectedReefCard";
import EnhancedTableHead from "./tableHead";
import ReefTableBody from "./body";
import { Order, OrderKeys } from "./utils";
import { reefsListLoadingSelector } from "../../../store/Reefs/reefsListSlice";

const ReefTable = ({ openDrawer, classes }: ReefTableProps) => {
  const loading = useSelector(reefsListLoadingSelector);
  const [order, setOrder] = useState<Order>("desc");
  const [orderBy, setOrderBy] = useState<OrderKeys>("alert");

  const handleRequestSort = (event: unknown, property: OrderKeys) => {
    const isAsc = orderBy === property && order === "asc";
    setOrder(isAsc ? "desc" : "asc");
    setOrderBy(property);
  };

  return (
    <>
      <Hidden smUp>
        <Grid container justify="center" style={{ marginBottom: "-3rem" }}>
          <Grid item>
            <IconButton>
              {openDrawer ? <ArrowDownwardIcon /> : <ArrowUpwardIcon />}
            </IconButton>
          </Grid>
        </Grid>
        {openDrawer ? null : (
          <Typography
            style={{
              position: "relative",
              margin: "1rem 0 0.5rem 1rem",
            }}
            variant="h5"
            color="textSecondary"
          >
            All Reefs
          </Typography>
        )}
      </Hidden>
      <SelectedReefCard />
      <Box
        className={classes.table}
        display="flex"
        flexDirection="column"
        flex={1}
      >
        <TableContainer>
          <Table stickyHeader>
            <EnhancedTableHead
              order={order}
              orderBy={orderBy}
              onRequestSort={handleRequestSort}
            />
            <ReefTableBody order={order} orderBy={orderBy} />
          </Table>
        </TableContainer>
        {loading && (
          <Box
            display="flex"
            flex={1}
            alignItems="center"
            justifyContent="center"
          >
            <CircularProgress size="10rem" thickness={1} />
          </Box>
        )}
      </Box>
    </>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    table: {
      paddingLeft: 10,
      height: "70%",
      overflowY: "auto",
      [theme.breakpoints.down("xs")]: {
        paddingLeft: 0,
      },
    },
  });

interface ReefTableIncomingProps {
  openDrawer: boolean;
}

type ReefTableProps = ReefTableIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(ReefTable);
