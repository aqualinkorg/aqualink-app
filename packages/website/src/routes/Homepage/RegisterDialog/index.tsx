import React, {
  BaseSyntheticEvent,
  useCallback,
  useState,
  useEffect,
} from "react";
import { Link } from "react-router-dom";
import {
  withStyles,
  WithStyles,
  createStyles,
  Dialog,
  Card,
  CardHeader,
  IconButton,
  CardContent,
  Typography,
  Grid,
  TextField,
  Button,
  Checkbox,
  LinearProgress,
  Collapse,
} from "@material-ui/core";
import { Alert } from "@material-ui/lab";
import CloseIcon from "@material-ui/icons/Close";
import { useForm } from "react-hook-form";
import { useDispatch, useSelector } from "react-redux";

import {
  createUser,
  userInfoSelector,
  userLoadingSelector,
  userErrorSelector,
} from "../../../store/User/userSlice";
import { UserRegisterParams } from "../../../store/User/types";
import incomingStyles from "../styles";

const RegisterDialog = ({
  open,
  handleRegisterOpen,
  handleSignInOpen,
  classes,
}: RegisterDialogProps) => {
  const dispatch = useDispatch();
  const user = useSelector(userInfoSelector);
  const loading = useSelector(userLoadingSelector);
  const error = useSelector(userErrorSelector);
  const [errorAlertOpen, setErrorAlertOpen] = useState<boolean>(false);
  const [readTerms, setReadTerms] = useState<boolean>(false);

  const { register, errors, handleSubmit } = useForm({
    reValidateMode: "onSubmit",
  });

  const onSubmit = useCallback(
    (
      data: any,
      event?: BaseSyntheticEvent<object, HTMLElement, HTMLElement>
    ) => {
      if (event) {
        event.preventDefault();
      }
      const registerInfo: UserRegisterParams = {
        fullName: `${data.firstName} ${data.lastName}`,
        organization: data.organization,
        email: data.emailAddress,
        password: data.password,
      };
      dispatch(createUser(registerInfo));
    },
    [dispatch]
  );

  useEffect(() => {
    if (user) {
      handleRegisterOpen(false);
    }
    if (error) {
      setErrorAlertOpen(true);
    }
  }, [user, handleRegisterOpen, error]);

  return (
    <Dialog open={open} maxWidth="md">
      <Card>
        <CardHeader
          className={classes.dialogHeader}
          title={
            <Grid container justify="center" item xs={12}>
              <Grid
                container
                alignItems="center"
                justify="space-around"
                item
                xs={12}
              >
                <Grid container item xs={6}>
                  <Typography variant="h4">Aqua</Typography>
                  <Typography
                    className={classes.dialogHeaderSecondPart}
                    variant="h4"
                  >
                    link
                  </Typography>
                </Grid>
                <Grid container justify="flex-end" item xs={1}>
                  <IconButton
                    className={classes.closeButton}
                    size="small"
                    onClick={() => handleRegisterOpen(false)}
                  >
                    <CloseIcon />
                  </IconButton>
                </Grid>
              </Grid>
            </Grid>
          }
        />
        {loading && <LinearProgress />}
        {error && (
          <Collapse in={errorAlertOpen}>
            <Alert
              severity="error"
              action={
                <IconButton
                  aria-label="close"
                  color="inherit"
                  size="small"
                  onClick={() => {
                    setErrorAlertOpen(false);
                  }}
                >
                  <CloseIcon fontSize="inherit" />
                </IconButton>
              }
            >
              {error}
            </Alert>
          </Collapse>
        )}
        <CardContent className={classes.contentWrapper}>
          <Grid container justify="center" item xs={12}>
            <Grid className={classes.dialogContentTitle} container item xs={10}>
              <Grid item>
                <Typography variant="h5" color="textSecondary">
                  Create an account
                </Typography>
              </Grid>
            </Grid>
            <Grid container item xs={10}>
              <form className={classes.form} onSubmit={handleSubmit(onSubmit)}>
                <Grid className={classes.textFieldWrapper} item xs={12}>
                  <TextField
                    id="firstName"
                    name="firstName"
                    placeholder="First Name"
                    helperText={
                      errors.firstName ? errors.firstName.message : ""
                    }
                    label="First Name"
                    inputRef={register({
                      required: "This is a required field",
                    })}
                    error={!!errors.firstName}
                    inputProps={{ className: classes.textField }}
                    fullWidth
                    variant="outlined"
                  />
                </Grid>
                <Grid className={classes.textFieldWrapper} item xs={12}>
                  <TextField
                    id="lastName"
                    name="lastName"
                    placeholder="Last Name"
                    helperText={errors.lastName ? errors.lastName.message : ""}
                    label="Last Name"
                    inputRef={register({
                      required: "This is a required field",
                    })}
                    error={!!errors.lastName}
                    inputProps={{ className: classes.textField }}
                    fullWidth
                    variant="outlined"
                  />
                </Grid>
                <Grid className={classes.textFieldWrapper} item xs={12}>
                  <TextField
                    id="organization"
                    name="organization"
                    placeholder="Organization"
                    helperText={
                      errors.organization ? errors.organization.message : ""
                    }
                    label="Organization"
                    inputRef={register({
                      required: "This is a required field",
                    })}
                    error={!!errors.organization}
                    inputProps={{ className: classes.textField }}
                    fullWidth
                    variant="outlined"
                  />
                </Grid>
                <Grid className={classes.textFieldWrapper} item xs={12}>
                  <TextField
                    id="emailAddress"
                    name="emailAddress"
                    placeholder="Email Address"
                    helperText={
                      errors.emailAddress ? errors.emailAddress.message : ""
                    }
                    label="Email Address"
                    inputRef={register({
                      required: "This is a required field",
                      pattern: {
                        value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
                        message: "Invalid email address",
                      },
                    })}
                    error={!!errors.emailAddress}
                    inputProps={{ className: classes.textField }}
                    fullWidth
                    variant="outlined"
                  />
                </Grid>
                <Grid className={classes.textFieldWrapper} item xs={12}>
                  <TextField
                    id="password"
                    name="password"
                    type="password"
                    placeholder="Password"
                    helperText={errors.password ? errors.password.message : ""}
                    label="Password"
                    inputRef={register({
                      required: "This is a required field",
                      minLength: {
                        value: 8,
                        message: "Password must be at least 8 characters",
                      },
                    })}
                    error={!!errors.password}
                    inputProps={{ className: classes.textField }}
                    fullWidth
                    variant="outlined"
                  />
                </Grid>
                <Grid container justify="space-around" item xs={11}>
                  <Grid item xs={1}>
                    <Checkbox
                      className={classes.termsCheckbox}
                      checked={readTerms}
                      onChange={() => setReadTerms(!readTerms)}
                      color="primary"
                    />
                  </Grid>
                  <Grid item xs={10}>
                    <Typography
                      className={classes.formText}
                      variant="subtitle1"
                      color="textSecondary"
                    >
                      I have read the{" "}
                      <Link className={classes.termsLink} to="/terms">
                        Terms and Conditions
                      </Link>
                    </Typography>
                  </Grid>
                </Grid>
                <Grid className={classes.button} item xs={12}>
                  <Button
                    fullWidth
                    type="submit"
                    color="primary"
                    variant="contained"
                    size="large"
                    disabled={!readTerms}
                  >
                    REGISTER NOW
                  </Button>
                </Grid>
                <Grid container item xs={12}>
                  <Typography
                    className={classes.formText}
                    variant="subtitle1"
                    color="textSecondary"
                  >
                    Have an account?{" "}
                    <Button
                      onClick={() => {
                        handleRegisterOpen(false);
                        handleSignInOpen(true);
                      }}
                      color="primary"
                    >
                      SIGN IN
                    </Button>
                  </Typography>
                </Grid>
              </form>
            </Grid>
          </Grid>
        </CardContent>
      </Card>
    </Dialog>
  );
};

const styles = () =>
  createStyles({
    ...incomingStyles,
    contentWrapper: {
      padding: 0,
    },
    termsCheckbox: {
      padding: 0,
      margin: "0 0 1rem 0",
    },
    termsLink: {
      color: "black",
    },
  });

interface RegisterDialogIncomingProps {
  open: boolean;
  handleRegisterOpen: (open: boolean) => void;
  handleSignInOpen: (open: boolean) => void;
}

type RegisterDialogProps = RegisterDialogIncomingProps &
  WithStyles<typeof styles>;

export default withStyles(styles)(RegisterDialog);
