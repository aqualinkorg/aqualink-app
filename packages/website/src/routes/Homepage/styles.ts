import theme from "../../layout/App/theme";

const styles = {
  closeButton: {
    color: theme.palette.primary.light,
  },
  dialogHeader: {
    backgroundColor: theme.palette.primary.main,
  },
  dialogHeaderSecondPart: {
    color: "#8AC6DE",
  },
  dialogContentTitle: {
    marginTop: "0.5rem",
  },
  form: {
    width: "100%",
  },
  textFieldWrapper: {
    margin: "1rem 0 1rem 0",
  },
  textField: {
    color: "black",
  },
  formText: {
    color: theme.palette.grey[500],
    fontWeight: 400,
  },
  button: {
    marginBottom: "0.5rem",
  },
};

export default styles;
