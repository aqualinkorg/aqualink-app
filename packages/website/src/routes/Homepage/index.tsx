import React, { useEffect, useState } from "react";
import { useDispatch } from "react-redux";
import {
  Grid,
  withStyles,
  WithStyles,
  createStyles,
  Hidden,
  Drawer,
} from "@material-ui/core";
import classNames from "classnames";

import HomepageNavBar from "../../common/NavBar";
import HomepageMap from "./Map";
import ReefTable from "./ReefTable";
import { reefsRequest } from "../../store/Reefs/reefsListSlice";

const Homepage = ({ classes }: HomepageProps) => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(reefsRequest());
  }, [dispatch]);

  const [openDrawer, setOpenDrawer] = useState(false);

  const toggleDrawer = () => {
    setOpenDrawer(!openDrawer);
  };

  return (
    <>
      <div role="presentation" onClick={openDrawer ? toggleDrawer : () => {}}>
        <HomepageNavBar searchLocation />
      </div>
      <div className={classes.root}>
        <Grid container direction="row">
          <Grid className={classes.map} item xs={12} sm={6}>
            <HomepageMap />
          </Grid>
          <Hidden xsDown>
            <Grid className={classes.reefTable} item sm={6}>
              <ReefTable openDrawer={openDrawer} />
            </Grid>
          </Hidden>
          <Hidden smUp>
            <Drawer
              variant="permanent"
              anchor="bottom"
              open={openDrawer}
              onClose={toggleDrawer}
              className={classNames({
                [classes.openDrawer]: openDrawer,
                [classes.closedDrawer]: !openDrawer,
              })}
              classes={{
                paper: classNames(classes.drawer, {
                  [classes.openDrawer]: openDrawer,
                  [classes.closedDrawer]: !openDrawer,
                }),
              }}
            >
              <div role="presentation" onClick={toggleDrawer}>
                <ReefTable openDrawer={openDrawer} />
              </div>
            </Drawer>
          </Hidden>
        </Grid>
      </div>
    </>
  );
};

const styles = () =>
  createStyles({
    root: {
      display: "flex",
      flexGrow: 1,
    },
    map: {
      display: "flex",
    },
    reefTable: {
      display: "flex",
      flexDirection: "column",
      height: "calc(100vh - 64px);", // subtract height of the navbar
    },
    drawer: {
      borderTopLeftRadius: "15px",
      borderTopRightRadius: "15px",
    },
    openDrawer: {
      height: "calc(95% - 64px);", // subtract height of the navbar
    },
    closedDrawer: {
      height: "50px",
      overflow: "revert",
    },
  });

type HomepageProps = WithStyles<typeof styles>;

export default withStyles(styles)(Homepage);
