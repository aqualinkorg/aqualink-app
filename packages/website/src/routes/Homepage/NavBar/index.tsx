import React, { useState, ChangeEvent } from "react";
import {
  AppBar,
  Toolbar,
  Grid,
  IconButton,
  Typography,
  InputBase,
  Avatar,
  Button,
  Menu,
  MenuItem,
  withStyles,
  WithStyles,
  createStyles,
  Theme,
} from "@material-ui/core";
import MenuIcon from "@material-ui/icons/Menu";
import SearchIcon from "@material-ui/icons/Search";
import NotificationsIcon from "@material-ui/icons/Notifications";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import { useSelector, useDispatch } from "react-redux";

import RegisterDialog from "../RegisterDialog";
import SignInDialog from "../SignInDialog";
import { userInfoSelector, signOutUser } from "../../../store/User/userSlice";

const HomepageNavBar = ({ classes }: HomepageNavBarProps) => {
  const user = useSelector(userInfoSelector);
  const dispatch = useDispatch();
  const [searchLocationText, setSearchLocationText] = useState<string>("");
  const [registerDialogOpen, setRegisterDialogOpen] = useState<boolean>(false);
  const [signInDialogOpen, setSignInDialogOpen] = useState<boolean>(false);
  const [anchorEl, setAnchorEl] = React.useState<HTMLElement | null>(null);

  const onChangeSearchText = (
    event: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
  ) => {
    setSearchLocationText(event.target.value);
  };

  // TODO: Dispatch action to search for reefs based on value
  const onSearchSubmit = () => {
    // eslint-disable-next-line no-console
    console.log(searchLocationText);
  };

  const handleRegisterDialog = (open: boolean) => setRegisterDialogOpen(open);
  const handleSignInDialog = (open: boolean) => setSignInDialogOpen(open);

  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleMenuClose = () => {
    setAnchorEl(null);
  };

  return (
    <>
      <AppBar className={classes.appBar} position="static" color="primary">
        <Toolbar>
          <Grid container alignItems="center" item xs={12}>
            <Grid item xs={1}>
              <IconButton edge="start" color="inherit">
                <MenuIcon />
              </IconButton>
            </Grid>
            <Grid container item xs={5}>
              <Typography variant="h4">Aqua</Typography>
              <Typography style={{ color: "#8AC6DE" }} variant="h4">
                link
              </Typography>
            </Grid>
            {user ? (
              <>
                <Grid container justify="flex-end" item xs={3}>
                  <Grid
                    className={classes.searchBar}
                    container
                    alignItems="center"
                    item
                    xs={8}
                  >
                    <Grid
                      className={classes.searchBarIcon}
                      item
                      xs={2}
                      container
                      alignItems="center"
                      justify="center"
                    >
                      <IconButton size="small" onClick={onSearchSubmit}>
                        <SearchIcon />
                      </IconButton>
                    </Grid>
                    <Grid
                      className={classes.searchBarText}
                      item
                      xs={10}
                      container
                      alignItems="center"
                    >
                      <InputBase
                        value={searchLocationText}
                        onChange={onChangeSearchText}
                        placeholder="Search location"
                      />
                    </Grid>
                  </Grid>
                </Grid>
                <Grid container justify="flex-end" item xs={3}>
                  <IconButton>
                    <NotificationsIcon />
                  </IconButton>
                  <IconButton>
                    <Avatar />
                  </IconButton>
                  <IconButton onClick={handleClick}>
                    <ExpandMoreIcon />
                  </IconButton>
                  <Menu
                    key="user-menu"
                    className={classes.userMenu}
                    anchorEl={anchorEl}
                    keepMounted
                    open={Boolean(anchorEl)}
                    onClose={handleMenuClose}
                  >
                    <MenuItem
                      className={classes.menuItem}
                      onClick={() => {
                        dispatch(signOutUser());
                        handleMenuClose();
                      }}
                    >
                      Logout
                    </MenuItem>
                  </Menu>
                </Grid>
              </>
            ) : (
              <Grid container justify="flex-end" item xs={6}>
                <Grid item>
                  <Button onClick={() => handleSignInDialog(true)}>
                    SIGN IN
                  </Button>
                </Grid>
                <Grid item>
                  <Button onClick={() => handleRegisterDialog(true)}>
                    REGISTER
                  </Button>
                </Grid>
              </Grid>
            )}
          </Grid>
        </Toolbar>
      </AppBar>
      <RegisterDialog
        open={registerDialogOpen}
        handleRegisterOpen={handleRegisterDialog}
        handleSignInOpen={handleSignInDialog}
      />
      <SignInDialog
        open={signInDialogOpen}
        handleRegisterOpen={handleRegisterDialog}
        handleSignInOpen={handleSignInDialog}
      />
    </>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    appBar: {
      height: 64,
      "&.MuiPaper-root": {
        backgroundColor: theme.palette.primary.main,
      },
    },
    searchBar: {
      height: 42,
    },
    searchBarIcon: {
      backgroundColor: "#6ba8c0",
      borderRadius: "4px 0 0 4px",
      height: "100%",
    },
    searchBarText: {
      paddingLeft: "0.5rem",
      backgroundColor: "#469abb",
      borderRadius: "0 4px 4px 0",
      height: "100%",
    },
    userMenu: {
      marginTop: "3rem",
    },
    menuItem: {
      margin: 0,
      backgroundColor: theme.palette.grey[500],
      color: theme.palette.text.secondary,
    },
  });

type HomepageNavBarProps = WithStyles<typeof styles>;

export default withStyles(styles)(HomepageNavBar);
