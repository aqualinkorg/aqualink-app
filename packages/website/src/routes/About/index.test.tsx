import React from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";

import Homepage from ".";

jest.mock("./ReefTable", () => "Mock-ReefTable");
const mockStore = configureStore([]);
describe("Homepage", () => {
  let element: HTMLElement;
  beforeEach(() => {
    const store = mockStore({
      reefsList: {
        list: [],
        loading: false,
        error: null,
      },
      homepage: {
        reefOnMap: null,
      },
      user: {
        userInfo: null,
        loading: false,
        error: null,
      },
    });

    store.dispatch = jest.fn();

    element = render(
      <Provider store={store}>
        <Router>
          <Homepage />
        </Router>
      </Provider>
    ).container;
  });

  it("should render with given state from Redux store", () => {
    expect(element).toMatchSnapshot();
  });
});
