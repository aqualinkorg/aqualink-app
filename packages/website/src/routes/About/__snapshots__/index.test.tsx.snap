// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should render with given state from Redux store 1`] = `
<div>
  <mock-navbar
    searchlocation="false"
  />
  <div>
    <div
      class="page-container"
    >
      <div
        class="bloc l-bloc"
        id="bloc-0"
      >
        <div
          class="bloc l-bloc"
          id="bloc-1"
        >
          <div
            class="container bloc-md mobilecentered"
          >
            <div
              class="row"
            >
              <div
                class="centered order-lg-0 order-1 order-md-0 order-sm-0 col"
              >
                <mock-typography
                  classname="About-title-1"
                  variant="h4"
                >
                  About Us
                </mock-typography>
                <p>
                  Aqualink is a philanthropic engineering organization working on Ocean conservation problems.
                  <br />
                </p>
              </div>
              <div
                class="offset-lg-0 col-lg-12 order-lg-1"
              >
                <img
                  alt="reef1b"
                  class="img-fluid mx-auto d-block imagepaddingtop img-responsive lazyload"
                  src="reef1b.jpg"
                />
              </div>
            </div>
            <div
              class="row voffset-lg"
            >
              <div
                class="centered order-lg-0 order-1 order-md-0 order-sm-0 col"
              >
                <mock-typography
                  classname="About-title-1"
                  variant="h4"
                >
                  The Team
                </mock-typography>
                <p>
                  We have been working in rapid development and entrepreneurial businesses that scale and hope to bring a similar mindset to Ocean conservation.
                  <br />
                  <br />
                  We have an extended team of engineering contractors in a variety of disciplines all of whom embrace open source philosophies and want to help build Ocean conservation tools.Â 
                  <br />
                </p>
              </div>
            </div>
            <div
              class="row voffset-md"
            >
              <div
                class="col-lg-4 col-md-4"
              >
                <div
                  class="card border-0"
                >
                  <div
                    class="card-body team-card"
                  >
                    <img
                      alt="placeholder user"
                      class="rounded-circle mx-auto d-block mt-5 img-style lazyload"
                      src="peter.jpg"
                      width="100"
                    />
                    <h5
                      class="text-center mg-sm"
                    >
                      Peter Rive
                    </h5>
                    <p
                      class="text-lg-center"
                    >
                      Co-founder of SolarCity, a pioneer in making solar energy an affordable alternative to fossil fuels for homeowners.
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="col-lg-4 col-md-4 mt-3 mt-md-0"
              >
                <div
                  class="card border-0"
                >
                  <div
                    class="card-body team-card"
                  >
                    <img
                      alt="placeholder user"
                      class="rounded-circle mx-auto d-block mt-5 img-placeholder-us-style lazyload"
                      src="LYNDON1.jpg"
                      width="100"
                    />
                    <h5
                      class="text-center mg-sm"
                    >
                      Lyndon Rive
                    </h5>
                    <p
                      class="text-lg-center"
                    >
                      Also a co-founder of SolarCity, current National Geographic board member, and member of the USA underwater hockey team.
                      <br />
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="col-lg-4 col-md-4 mt-3 mt-md-0"
              >
                <div
                  class="card border-0"
                >
                  <div
                    class="card-body team-card"
                  >
                    <img
                      alt="placeholder user"
                      class="rounded-circle mx-auto d-block mt-5 img-3-style lazyload"
                      src="drew.jpg"
                      width="100"
                    />
                    <h5
                      class="text-center mg-sm"
                    >
                      Drew Gray
                    </h5>
                    <p
                      class="text-lg-center"
                    >
                      Engineer and executive with experience building autonomous systems at Tesla, Cruise, Uber, and Voyage. Instrumental in pushing the state-of-the-art in self-driving cars using computer vision and deep learning.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <mock-footer />
</div>
`;
